### PR TITLE
feat:     A new feature(engine): Added fuzzy search feature

### PR DESCRIPTION
**Pull request by SirWindfield**
This makes searching for commit types possible:

### CODE REVIEW

File: engine.js [hunk 1]
```diff
@@ -26,6 +30,18 @@ module.exports = function (options) {
...
 
+  var searchTypes = (answers, input) => {
+    input = input || '';
+    return new Promise(function(resolve) {
+      setTimeout(function() {
+        const filtered = choices
+                          .filter(a => a.value.toLowerCase().includes(input))
+                          .map(a => a.name);
+        resolve(filtered);
+        }, 150);
```
Comment (LinusU) [inline engine.js]: 🤔

Comment (SirWindfield) [reply engine.js]: Oh, I forgot to remove the `randomInt` method. I used to to feed a delay into the `timeout` call so it doesn't "feel the same" if you search each time. Is the rest ok though?

Comment (SirWindfield) [reply engine.js]: And I actually broke the commit message while implementing it...

Comment (SirWindfield) [reply engine.js]: I'll clean it up and re-open...

**Pull request closed without merging**